core/src/main/java/org/ehcache/EhcacheManager.java

```diff
@@ -300,9 +302,9 @@ public void release() {
       ServiceLocator.findAmongst(CacheEventListenerConfiguration.class, config.getServiceConfigurations().toArray());
       for (CacheEventListenerConfiguration lsnrConfig: evtLsnrConfigs) {
         // XXX this assumes a new instance returned for each call - yet args are always the same. Is this okay?
-        final CacheEventListener<K, V> lsnr = evntLsnrFactory.createEventListener(alias, config);
-        evtService.registerCacheEventListener(lsnr, lsnrConfig.orderingMode(), lsnrConfig.firingMode(),
-            lsnrConfig.fireOn());
+        final CacheEventListener<K, V> lsnr = evntLsnrFactory.createEventListener(alias, lsnrConfig);
+          ehCache.getRuntimeConfiguration().registerCacheEventListener(lsnr, lsnrConfig.orderingMode(), lsnrConfig.firingMode(),
+          lsnrConfig.fireOn());
         if (lsnr != null) {
```

Comment:
There is something fishy about this null check. If for whatever reason the CacheEventListenerFactory creates a null listener, we do register it in the cache's runtime config but not in the releasables stack.

Either creating a null listener makes sense and the null check should also encompass the registration in the runtime config, or that doesn't make sense and the null check should be done right after creation and throw an exception.